File: PHP/Challenge/September/12_Count_the_Number_of_Consistent_Strings.php
<?php
// https://leetcode.com/problems/count-the-number-of-consistent-strings/description/?envType=daily-question&envId=2024-09-12
function countConsistentStrings($allowed, $words)
{
    $count = 0;
    foreach ($words as $word) {
        $count ++;
        foreach (str_split($word) as $letter) {
            if (strpos($allowed, $letter) === false) {
                $count --;
                break;
            }
        }
    }
    return $count;
}

File: Python/Challenge/September/12_Count_the_Number_of_Consistent_Strings.py
# https://leetcode.com/problems/count-the-number-of-consistent-strings/description/?envType=daily-question&envId=2024-09-12

def countConsistentStrings(allowed: str, words: list[str]) -> int:
    count = 0
    for word in words:
        count += 1
        for letter in word:
            if letter not in allowed:
                count -= 1
                break
    return count


allowed = "ab"
words = ["ad", "bd", "aaab", "baa", "badab"]
print(countConsistentStrings(allowed, words))
